### PR TITLE
Dataframe

### PR DESCRIPTION
**Pull request by julian-evers**

### CODE REVIEW

**Comment by rtimms** [comment]
thanks, looks good. you could make it cleaner by doing something like 
```python3
parameters = [
    "Volumetric stack energy density [Wh.L-1]",
    "Gravimetric stack energy density [Wh.kg-1]",
]
names = []
units = []
values = []
for parameter in parameters:
    name, unit = parameter.split("[")
    names.append(name.strip())
    units.append(unit.rstrip(']'))
    values.append(stack_ed[parameter])

data = {
    "Parameter": names,
    "Unit": units,
    "Value": values,
}
```
this would require you to have all the values in the units you want in the breakdown dicts, but i think it's cleaner as you don't need to manually specific a name and units separately - they are just taken from the key of the dictionary

File: pybamm_tea/tea.py [hunk 1]
```diff
@@ -333,209 +333,155 @@ def print_stack_breakdown(self):
...
-                "Parameter": "Positive electrode stoichiometry at 0% SoC",
-                "Unit": "-",
-                "Value": stack_ed.get("Positive electrode stoichiometry at 0% SoC"),
-            },
-        ]
-        # Create the DataFrame from the list of dictionaries
+        
+        data = {
+            "Parameter": [
+                "",
```
Comment (rtimms) [inline pybamm_tea/tea.py]: why the empty entries?

Comment (julian-evers) [reply pybamm_tea/tea.py]: They were unintended.

File: pybamm_tea/tea.py [hunk 2]
```diff
@@ -333,209 +333,102 @@ def print_stack_breakdown(self):
...
+
+        names = []
+        units = []
+        values = []
+        for parameter in parameters:
+            if "[" in parameter:
+                name, unit = parameter.split("[")
+                names.append(name.strip())
+                units.append(unit.rstrip(']'))
+            else:
```
Comment (rtimms) [inline pybamm_tea/tea.py:427]: why do you need this? I don't think you need to have the subheadings as part of the dataframe

File: pybamm_tea/tea.py [hunk 3]
```diff
@@ -333,209 +333,102 @@ def print_stack_breakdown(self):
...
+                "mA.h.cm-2",
+                "um",
+                "kg.L-1",
+            ],
+            "Value": [
+                stack_ed.get("Volumetric stack energy density [W.h.L-1]"),
+                stack_ed.get("Gravimetric stack energy density [W.h.kg-1]"),
+                stack_ed.get("Stack average OCP [V]"),
+                stack_ed.get("Capacity [mA.h.cm-2]"),
+                10**6 * stack_ed.get("Stack thickness [m]"),
```
Comment (rtimms) [inline pybamm_tea/tea.py]: Just add "Stack thickness [um]" and "Stack density [kg.L-1]" to `stack_ed` too when you construct it, then you can just do the loop here too

**Review by rtimms — COMMENTED** [review]
thanks, looks good! Just a couple of minor changes and need to fix the merge conflicts then it's good to go

File: pybamm_tea/tea.py [hunk 4]
```diff
@@ -395,172 +388,62 @@ def print_capacities_and_potentials(self):
...
+            "Positive electrode stoichiometry at 100% SoC",
+            "Positive electrode stoichiometry at 0% SoC",
         ]
-        # Create the DataFrame from the list of dictionaries
+
+        names = []
+        units = []
+        values = []
+        for parameter in parameters:
+            if "[" in parameter:
```
Comment (rtimms) [inline pybamm_tea/tea.py:423]: alternatively, you could denote no units by "[-]"

**Review by rtimms — APPROVED** [review]

**Pull request merged**